presto-hive/src/main/java/com/facebook/presto/hive/HiveSplitManager.java

```diff
@@ -248,14 +254,16 @@ public CounterStat getHighMemorySplitSource()
                 }
 
                 // verify all partition is online
-                String protectMode = partition.getParameters().get(ProtectMode.PARAMETER_NAME);
-                String partName = makePartName(table.getPartitionColumns(), partition.getValues());
-                if (protectMode != null && getProtectModeFromString(protectMode).offline) {
-                    throw new PartitionOfflineException(tableName, partName, false, null);
-                }
-                String prestoOffline = partition.getParameters().get(PRESTO_OFFLINE);
-                if (!isNullOrEmpty(prestoOffline)) {
-                    throw new PartitionOfflineException(tableName, partName, true, prestoOffline);
+                verifyOnline(
+                        tableName,
+                        Optional.of(makePartName(table.getPartitionColumns(), partition.getValues())),
+                        getProtectMode(partition),
+                        table.getParameters());
+
+                // verify all partition is readable
+                String partitionNotReadable = partition.getParameters().get(OBJECT_NOT_READABLE);
+                if (!isNullOrEmpty(partitionNotReadable)) {
+                    throw new HiveNotReadableException(tableName, Optional.of(partName), partitionNotReadable);
```

Comment:
will the exception message include table name and partition name?

Reply:
Fixed! Thank you !